Makefile

```diff
@@ -0,0 +1,62 @@
+# Copyright 2021 The kube-storage Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+SHELL:=/bin/bash
+
+PROTOC_REQ := "libprotoc 3.14.0"
+PROTOC_GEN_GO_REQ := "protoc-gen-go v1.25.0"
+PROTOC_GEN_GO_GRPC_REQ := "protoc-gen-go-grpc 1.1.0"
+
+PROTOC := $(shell ([ "`$(PWD)/protoc --version 2> /dev/null`" == $(PROTOC_REQ) ] && echo true))
+PROTOC_GEN_GO := $(shell ([ "`$(PWD)/protoc-gen-go --version 2> /dev/null`" == $(PROTOC_GEN_GO_REQ) ] && echo true))
+PROTOC_GEN_GO_GRPC := $(shell ([ "`$(PWD)/protoc-gen-go-grpc --version 2> /dev/null`" == $(PROTOC_GEN_GO_GRPC_REQ) ] && echo true))
+
+all: install-deps build
+
+build: install-deps
+	# generate libs
+	./protoc --go_out=lib/go/replication --go_opt=paths=source_relative --plugin=./protoc-gen-go *.proto
+	./protoc --go-grpc_out=lib/go/replication --go-grpc_opt=paths=source_relative --plugin=./protoc-gen-go-grpc *.proto
+
+install-deps:
+ifndef PROTOC
+	# download protoc
+	curl -OL https://github.com/protocolbuffers/protobuf/releases/download/v3.14.0/protoc-3.14.0-linux-x86_64.zip
+	unzip -jod . protoc-3.14.0-linux-x86_64.zip bin/protoc
+
+	# extract include/google/protobuf/descriptor.proto
+	mkdir -p google/protobuf
+	unzip -jod google/protobuf protoc-3.14.0-linux-x86_64.zip include/google/protobuf/descriptor.proto
+endif
+
+ifndef PROTOC_GEN_GO
+	# download protoc-gen-go
+	curl -OL https://github.com/protocolbuffers/protobuf-go/releases/download/v1.25.0/protoc-gen-go.v1.25.0.linux.386.tar.gz
+	tar -zxvf protoc-gen-go.v1.25.0.linux.386.tar.gz protoc-gen-go -C ./
+endif
+
+ifndef PROTOC_GEN_GO_GRPC
+	# download protoc-gen-go-grpc
+	curl -OL https://github.com/grpc/grpc-go/releases/download/cmd%2Fprotoc-gen-go-grpc%2Fv1.1.0/protoc-gen-go-grpc.v1.1.0.linux.386.tar.gz
+	tar -zxvf protoc-gen-go-grpc.v1.1.0.linux.386.tar.gz ./protoc-gen-go-grpc -C ./
+endif
+
+clean:
+	rm -rf protoc \
+		protoc-gen-go \
+		protoc-gen-go-grpc \
+		protoc-3.14.0-linux-x86_64.zip \
+		protoc-gen-go.v1.25.0.linux.386.tar.gz \
+		protoc-gen-go-grpc.v1.1.0.linux.386.tar.gz \
+		google/protobuf/*.proto
```

Comment:
indentation? and also its good to install these packages to a folder and remove the folder